lighthouse-core/audits/time-to-interactive.js

```diff
@@ -0,0 +1,129 @@
+   *     Alternative definition (from Chrome on Android Progress Bar Enhancements - google-only, sry): top frame's DOMContentLoaded + top frame's images (who started before DCL) are loaded
+   *
+   * The page is actually ready for user:
+   *   - domContentLoadedEventEnd has fired
+   *     Definition: HTML parsing has finished, all DOMContentLoaded handlers have run.
+   *     No risk of DCL event handlers changing the page
+   *     No surprises of inactive buttons/actions as DOM element event handlers should be bound
+   *   - The main thread is available enough to handle user input
+   *     first 500ms window where Est Input Latency is <50ms at the 90% percentile.
+   * @param {!Artifacts} artifacts The artifacts from the gather phase.
+   * @return {!AuditResult} The score from the audit, ranging from 0-100.
+   */
+  static audit(artifacts) {
+    // Max(FMPMetric, DCLEnded, visProgress[0.85]) is where we begin looking
+    return FMPMetric.audit(artifacts).then(fmpResult => {
+      const fmpTiming = fmpResult.rawValue;
+      const timings = fmpResult.extendedInfo.timings;
+
+      // process the trace
+      const tracingProcessor = new TracingProcessor();
+      const model = tracingProcessor.init(artifacts.traceContents);
+      const endOfTraceTime = model.bounds.max;
+
+      // todo DCLEnded
+      // TODO: Consider UA loading indicator
+
+      // look at speedline results for 85% starting at FMP
+      const visualProgress = artifacts.Speedline.frames.map(frame => {
+        return {
+          progress: frame.getProgress(),
+          time: frame.getTimeStamp()
+        };
+      });
+      const fMPts = timings.fMPfull + timings.navStart;
+      const visuallyReady = visualProgress.find(frame => {
+        return frame.time >= fMPts && frame.progress >= 85;
+      });
+      const visuallyReadyTiming = visuallyReady.time - timings.navStart;
+
+      // Find first 500ms window where Est Input Latency is <50ms at the 90% percentile.
+      let startTime = Math.max(fmpTiming, visuallyReadyTiming) - 50;
+      let endTime;
+      let currentLatency = Infinity;
+      const percentile = 0.9;
+      const threshold = 50;
+      let foundLatencies = [];
+
+      // When we've found a latency that's good enough, we're good.
+      while (currentLatency > threshold) {
+        // While latency is too high, increment just 50ms and look again.
+        startTime += 50;
+        endTime = startTime + 500;
+        // If there's no more room in the trace to look, we're done.
+        // TODO return an error instead
+        if (endTime > endOfTraceTime) {
+          return;
+        }
+        // Get our expected latency for the time window
+        const latencies = TracingProcessor.getRiskToResponsiveness(
+          model, artifacts.traceContents, startTime, endTime);
+        const estLatency = latencies.find(res => res.percentile === percentile);
```

Comment:
I added a `percentiles` argument, so you can pass in `[0.9]` as the last argument to `getRiskToResponsiveness()` and only get that in the results. Not much of an optimization, but you can avoid the `find` now :)